File: WappChatAnalyzer-frontend/src/app/dtos/emojiInfoTotal.ts
import { Sender } from "./sender";

export class EmojiInfoTotal {
    public emojiInfos: EmojiInfo[];
}

export class EmojiInfo {
    public emojiCodePoints: string;
    public total: number;
    public senders: { [Key: number]: Sender}
    public bySenders: { [Key: number]: number };
}

File: WappChatAnalyzer/DTOs/EmojiInfoTotal.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WappChatAnalyzer.DTOs
{
    public class EmojiInfoTotal
    {
        public List<EmojiInfo> EmojiInfos { get; set; }
    }

    public class EmojiInfo
    {
        public string EmojiCodePoints { get; set; }
        public int Total { get; set; }
        public Dictionary<int, SenderDTO> Senders { get; set; }
        public Dictionary<int, int> BySenders { get; set; }
    }
}
